pkg/sr/api.go

```diff
@@ -352,18 +352,32 @@ func (cl *Client) Schemas(ctx context.Context, subject string) ([]SubjectSchema,
 //
 // This supports param [Normalize].
 func (cl *Client) CreateSchema(ctx context.Context, subject string, s Schema) (SubjectSchema, error) {
+	return cl.CreateSchemaWithID(ctx, subject, s, -1)
+}
+
+// CreateSchemaWithID attempts to create a schema with a fixed ID in the given
+// subject. If the id is set to -1, this method is equivalent to CreateSchema().
+//
+// This supports param [Normalize].
+func (cl *Client) CreateSchemaWithID(ctx context.Context, subject string, s Schema, id int) (SubjectSchema, error) {
 	// POST /subjects/{subject}/versions => returns ID
 	// Newer SR returns the full SubjectSchema, but old does not, so we
 	// re-request to find the full information.
 	path := pathSubjectWithVersion(subject)
-	var id struct {
+	var into struct {
 		ID int `json:"id"`
 	}
-	if err := cl.post(ctx, path, s, &id); err != nil {
-		return SubjectSchema{}, err
+	if id == -1 {
+		if err := cl.post(ctx, path, s, &into); err != nil {
+			return SubjectSchema{}, err
+		}
+	} else {
+		if err := cl.post(ctx, path, SubjectSchema{Schema: s, ID: id}, &into); err != nil {
```

Comment:
Is this a RP specific thing, or the API in general?

Reply:
I found it in the Confluent docs here: https://docs.confluent.io/platform/current/schema-registry/installation/migrate.html#migrate-individual-schemas. See how they POST the schema ID. It’s not mentioned in the API reference though…